### PR TITLE
refactor league_stat to init with collections, add count_of_teams to …

### PR DESCRIPTION
**Pull request by elguapogordo**
refactor league_stat to init with collections, add count_of_teams to stat_tracker

### CODE REVIEW

File: lib/league_stat.rb [hunk 1]
```diff
@@ -3,16 +3,17 @@ class LeagueStat
   attr_reader :stats_by_team
 
   def initialize(teams_list, games_list)
+    @team_collection = TeamCollection.new(teams_list)
+    @game_collection = GameCollection.new(games_list)
```
Comment (elguapogordo) [inline lib/league_stat.rb:7]: this has been added to initialization of instance of LeagueStat

File: lib/league_stat.rb [hunk 2]
```diff
@@ -3,16 +3,17 @@ class LeagueStat
...
     @stats_by_team = Hash.new do |hash, key|
       hash[key] = Hash.new { |hash, key| hash[key] = 0 }
     end
-    create_teams(teams_list)
-    create_league_stats(games_list)
+    create_teams(@team_collection.teams_list)
+    create_league_stats(@game_collection.games_list)
```
Comment (elguapogordo) [inline lib/league_stat.rb:12]: updated to allow current methods to function

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -128,4 +129,14 @@ def fewest_tackles(season_param)
...
+  def count_of_teams
+    # require 'pry'; binding.pry
+    # games_file_path = './data/games.csv'
+    # teams_file_path = './data/teams.csv'
+    # @team_collection = TeamCollection.new(teams_file_path)
+    # @game_collection = GameCollection.new(games_file_path)
+    @league_stat = LeagueStat.new(@teams_file, @games_file)
+    @league_stat.count_of_teams
+  end
+
```
Comment (elguapogordo) [inline lib/stat_tracker.rb:141]: forgot to clean this up. will do that immediately as i work into the next method.

File: lib/league_stat.rb [hunk 4]
```diff
@@ -49,5 +50,3 @@ def count_of_teams
   end
 
 end
```
Comment (StephanieFriend) [inline lib/league_stat.rb:52]: Thanks for taking out the prys!

**Review by StephanieFriend — APPROVED** [review]
Overall looks great! Just don't forget to take out the comments. :)

**Pull request merged**